DRYD-1246: UCB Held In Trust Contribution

mikejritter opened the pull request:
**What does this do?**
* Adds the Held in Trust procedure to the services layer

**Additional Changes**
I applied some basic formatting and linting to the initial contribution. For the xml files I tried to keep things consistent with what we see in cspace-ui (e.g. 2 spaced indents), and for java I mostly used basic suggestions from intellij, including their formatting. I'll note that for formatting I've been keeping things consistent within the file, e.g. if the file had a majority tabs I would format with tabs. I believe for HIT everything ended up with spaces.

* Applied xml formatting
* Applied java formatting
* Removed log level checks
* Switched to parameterized logging
* Always iterate through ref items
  * Because of the log level checks going away these were always being run, so this was just removing an unnecessary conditional check
* Removed unused fields/variables

**Why are we doing this? (with JIRA link)**
Jira: https://collectionspace.atlassian.net/browse/DRYD-1246

This is part of the UCB contribution branch, and the Held in Trust procedure is something which was requested for an earlier version, though never brought in as it got stuck in the bundle. 

**How should this be tested? Do these changes have associated tests?**
The services and application layer can be quickly tested by running the integration test suite:
* Build with the corresponding PR from the application layer and the necessary tenants enabled to run the Integration Tests
* Start collection space
* Run the integration test suite

Further testing can be done with in the PR for cspace-ui which will also check the new endpoints can save and retrieve data. 

**Dependencies for merging? Releasing to production?**
* Application layer changes: https://github.com/collectionspace/application/pull/266
* UI layer changes: https://github.com/collectionspace/cspace-ui.js/pull/143

**Has the application documentation been updated for these changes?**
No

**Did someone actually run this code to verify it works?**
@mikejritter ran the integration tests

ray-lee commented:
I'm unsure about naming this "Hit". Isn't it confusing? Especially in this context:

```
$ curl -s -u admin@core.collectionspace.org:Administrator http://localhost:8180/cspace-services/servicegroups/procedure -H 'Accept: application/json' | jq
{
  "document": {
    "@name": "servicegroups",
    "ns2:servicegroups_common": {
      "@xmlns:ns2": "http://collectionspace.org/services/servicegroup",
      "name": "procedure",
      "uri": "/servicegroups/procedure",
      "hasDocTypes": {
        "hasDocType": [
          "Insurance",
          "Loanin",
          "Uoc",
          "Hit",
          "Group",
          "Osteology",
          "Conditioncheck",
          "Conservation",
          "Valuationcontrol",
          "ObjectExit",
          "Loanout",
          "Media",
          "Acquisition",
          "Intake",
          "Exhibition",
          "Transport",
          "Iterationreport",
          "Movement"
        ]
      }
    }
  }
}
```

Is it just for brevity, or is there another reason? We already have longer names than "HeldInTrust", which would be clearer.

ray-lee commented on services/hit/service/pom.xml:
```diff
@@ -0,0 +1,95 @@
+        </dependency>
+        <dependency>
+            <groupId>org.collectionspace.services</groupId>
+            <artifactId>org.collectionspace.services.collectionobject.jaxb</artifactId>
```
Is this dependency needed?

mikejritter replied:
I don't see any usages in the module, will remove and test.

ray-lee reviewed (changes requested):
Some general changes that will be needed:

- Change `hit` to `heldInTrust` (or "Held-in-trust" where appropriate). The case may vary depending on context (e.g. Java class name, file name, Nuxeo doctype name, URL path, plain English). Try to follow the prevailing convention for each type of usage.
- Nuxeo bundles are not being removed from tomcat/nuxeo-server/plugins when `ant undeploy` is run. Need to update services/build.xml, as in this example: https://github.com/mikejritter/services/commit/999942140b52fa15a3d667852b7901d52cee475b#diff-138f2342f3b61e75049507ea74f8fd91dd057523a75dc21310c58f9bb1529fba
- There are a lot of references to intake in the code, presumably left-over from copy/paste. I noted the places I saw them, but this whole PR should be searched for "intake" (case insensitively) to double check.

Other comments below.

mikejritter commented on services/hit/client/temp-testng-customsuite.xml:
Looks like it's from an old jboss example. Will get it removed.

mikejritter replied:
fyi @ray-lee I also removed duplicates of this file in dimension and intake.

mikejritter commented:
@ray-lee I _think_ I got all the changes:
* Moved from hit to heldintrust
  * Heldintrust for class names
  * heldintrust for the module/package name 
* Removed references to intake
```
[heldintrust (ucb-hit-refactor)] $ rg --include-zero --count -t xml 'intake|hit' 
3rdparty/build.xml:0
3rdparty/nuxeo-platform-cs-heldintrust/build.xml:0
3rdparty/nuxeo-platform-cs-heldintrust/pom.xml:0
3rdparty/pom.xml:0
build.xml:0
client/pom.xml:0
client/src/main/resources/log4j2-surefire.xml:0
client/src/test/resources/log4j2-surefire.xml:0
jaxb/pom.xml:0
jaxb/src/main/resources/heldintrusts-common.xsd:0
pom.xml:0
service/pom.xml:0
``` 
* Update variable names/parameters in tests to match their fields in Held-in-Trust
* Cleaned up dependencies
* Removed unused customsuite file
* Added heldintrust to build.xml deploy/undeploy/dist targets

I also merged in master in order to test everything against the most recent changes w/ SSO, which was more for convenience than anything. Integration tests for the heldintrust module pass, however the IntegrationTest module has been failing for me. I believe it was one of the xml replay tests, though it was hard to tell what test exactly was failing.

ray-lee reviewed (approved):
Looks good!
